docs/src/pages/components/data-grid/events/events.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Data Grid - Events
+components: DataGrid, XGrid
+---
+
+# Data Grid - Events
+
+<p class="description">The data grid emits events that can be subscribed to attach custom behavior.</p>
+
+## Subscribing to events
+
+You can subscribe to one of the [events emitted](/components/data-grid/events/#catalog-of-events) by calling `apiRef.current.subscribeEvent()` with the name of the event and a handler. The handler will be called with two arguments: a object with information related to the event and, optionally, a `React.SyntheticEvent` object if the event was emitted by a DOM element.
+
+The demo shows how to subcribe to the `columnResize` event. Try it by resizing the columns.
+
+```tsx
+/**
+ * Allows to register a handler for an event.
+ * @param event The name of event
+ * @param handler The handler to be called
+ * @param options Additional options for this listener
+ * @returns A function to unsubscribe from this event
+ */
+subscribeEvent: (
+  event: string,
+  handler: (params: any, event?: React.SyntheticEvent) => void,
+  options?: GridSubscribeEventOptions,
+) => () => void;
+```
+
+{{"demo": "pages/components/data-grid/events/SubscribeToEvents.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
Was the plan to use `GRID_COLUMN_RESIZE` in the demos or `columnResize`? I can't remember.

Reply:
Ok, cc @mui-org/x to make sure we all agree on the approach. I personally don't really mind.

The advantage of `GRID_COLUMN_RESIZE` is IMHO to be easier to search, but since `apiRef.current.subscribe` is easy to grep in a codebase, I don't see it really important.